### ISSUE TITLE
Change user's role after verification

### ISSUE THREAD

**Issue by judgej** [jrean/laravel-user-verification#8]
More a question than anything.

When users register, I am happy for them to be logged in immediately. However, some areas of the application need the user to be verified first, and I intend to implement that by assigning a role (which is checked by the authorisation layer).

I am guessing there are events that are fired which would allow me to update the user (giving them a role)?

This verification confirms the user's email is working. Changing their email would set them back to an unverified state again. That's fine.

How about checking the user is a human as a part of verification? Does this package support hooking in, say, a reCaptcha to do that? Or is that just something to add to the view and controller/middleware?

**Comment by judgej** [comment]
Sorry if these seem dumb or obvious questions, but trying to track down just where functionality lives in Laravel these days is become a lot more difficult, as traits inject stuff all over the place, and exceptions are thrown and caught in many far-apart places in the normal flow of the page now.

**Comment by jrean** [comment]
Hey @judgej thank you for the feedback! There are no dumb questions at all! By sharing your thoughts you just made me think I should add and update the package to manage the flow of resetting an email ... About the captcha and Co., I reckon it's not part neither the goal of this package.

Cheers

**Comment by jrean** [comment]
@judgej 

> This verification confirms the user's email is working. Changing their email would set them back to an unverified state again. That's fine.

I just updated the package. Now using the `generate()` method will automatically reset the state and so even if a user reset it's email one can just call again the `generate()` method to restart the verification process.

Cheers.

**Comment by judgej** [comment]
Just some background to what I was implementing yesterday, in case it is useful as a use-case:

I have the a `VerifyController` separate from the `AuthController`, as they do not operate together. Users registering is handled by the `AuthController` using the standard L5.1 approach, with users instantly logged in when they register. This gives them an account so gets them up and running quickly with a low barrier to entry.

Next the `VerifyController` is there to allow users to verify their account. They can do this any time they like. It consists of a reCaptcha first, to confirm they are human, then a verification token is sent out to confirm their email is valid.

Next I will create a middleware `verify` check that confirms whether their account is verified. If they try to access an area of the site that needs verification, then they will be taken to the verify page, have to go through the verify process, then are taken back to the page they tried to access.

So we have registered users who can log in an update their details, but can do relatively little else. Then we have verified users who get access to the remainder of the application functionality. It gets users on-board with the minimum of fuss when registering, but protects more of the application from bots and similar.

That's different to your example, where the verification is tagged onto the end of the registration process, logging the user in only when their account is verified. That's cool too, and it's good that with registration and verification kept separate, we can choose how to join them together.
